extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java

```diff
@@ -157,41 +159,10 @@ private static void setGrantClientParams(OidcClientConfig oidcConfig, MultiMap g
     }
 
     private static Uni<String> discoverTokenRequestUri(WebClient client, String authServerUrl, OidcClientConfig oidcConfig) {
-        final long connectionRetryCount = OidcCommonUtils.getConnectionRetryCount(oidcConfig);
-        if (connectionRetryCount > 1) {
-            LOG.infof("Connecting to IDP for up to %d times every 2 seconds", connectionRetryCount);
-        }
-
-        for (long i = 0; i < connectionRetryCount; i++) {
-            try {
-                if (oidcConfig.discoveryEnabled) {
-                    return discoverTokenEndpoint(client, authServerUrl);
-                }
-                break;
-            } catch (Throwable throwable) {
-                while (throwable instanceof CompletionException && throwable.getCause() != null) {
-                    throwable = throwable.getCause();
-                }
-                if (throwable instanceof OidcClientException) {
-                    if (i + 1 < connectionRetryCount) {
-                        try {
-                            Thread.sleep(2000);
-                        } catch (InterruptedException iex) {
-                            // continue connecting
-                        }
-                    } else {
-                        throw (OidcClientException) throwable;
-                    }
-                } else {
-                    throw new OidcClientException(throwable);
-                }
-            }
-        }
-        return Uni.createFrom().nullItem();
-    }
-
-    private static Uni<String> discoverTokenEndpoint(WebClient client, String authServerUrl) {
-        String discoveryUrl = authServerUrl + "/.well-known/openid-configuration";
+        final String discoveryUrl = authServerUrl + "/.well-known/openid-configuration";
```

Comment:
@Yelzhasdev Can you push this code to `OidcCommonUtils` - it may already have this check there - if not - please remove other connection-delay related code from `OidcCommonUtils`

Reply:
Yep, sure

Reply:
fixed

Reply:
@Yelzhasdev, thanks - note there is another function returning it in seconds just above there :-), so can you reuse that one and converts secs to millisecs here ?